packages/grzctl/src/grzctl/commands/consent.py

```diff
@@ -81,36 +76,18 @@ def _gather_consent_information(metadata: GrzSubmissionMetadata) -> dict[str, bo
     for donor in metadata.donors:
         for research_consent in donor.research_consents:
             mii_consent = research_consent.scope
-            if isinstance(mii_consent, str):
-                mii_consent = json.loads(mii_consent)
-            mii_consent = typing.cast(dict[str, Any], mii_consent)
-
-            if top_level_provision := mii_consent.get("provision"):
-                if top_level_provision.get("type") != FhirProvision.DENY:
-                    sys.exit(
-                        f"The root provision type must be deny, not {top_level_provision.get('type')}, "
-                        f"since the profile follows an opt-in consent scheme. "
-                        f"Explicit opt-in consents must be made via nested provisions."
-                    )
-                else:
-                    nested_provisions = top_level_provision.get("provision")
-                    consents[donor.donor_pseudonym] = _check_nested_provisions(nested_provisions)
+            if top_level_provision := mii_consent.provision:
+                nested_provisions = top_level_provision.provision
+                consents[donor.donor_pseudonym] = _check_nested_provisions(nested_provisions)
```

Comment:
So currently a patient is considered consented based on whatever the last consent stated, not sure if we want this behavior but it was the prior behavior as well

Reply:
Technically we'd have to check the dates I think? I do not know in which cases one donor has multiple research consent entries anyway. I think that's what we have to find out.